### PR TITLE
[api] Added api dependency in pc files

### LINKED ISSUE #1250: API dependency missing in pkg-config
API dependency on ml-api-common is missing in pkg config.

### PR DESCRIPTION
**Pull request by kparichay**
Added API dependecy in pc files for the capi and ccapi.
The dependency is of the capi-ml-common which provides ml-api-common.h header.

Resolves #1250

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Review by zhoonit — APPROVED** [review]
Nice :)

**Review by myungjoo — COMMENTED** [review]
You may need to update ```meson_options.txt``` so that its default behavior becomes compatible with non Tizen machines using api/capi/include/platform/ml-api-common.h.

**Comment by kparichay** [comment]
> You may need to update `meson_options.txt` so that its default behavior becomes compatible with non Tizen machines using api/capi/include/platform/ml-api-common.h.

I am not sure I understand your concern correctly. 
For non-tizen machines (like debian), the dependency defaults to `capi-ml-common` which is the pkgconfig name for ml-api-common package. Can you please elaborate your concern?

**Comment by myungjoo** [comment]
> > You may need to update `meson_options.txt` so that its default behavior becomes compatible with non Tizen machines using api/capi/include/platform/ml-api-common.h.
> 
> I am not sure I understand your concern correctly.
> For non-tizen machines (like debian), the dependency defaults to `capi-ml-common` which is the pkgconfig name for ml-api-common package. Can you please elaborate your concern?

If you are using capi-ml-common.pc of ml-api-common-dev debian package of github.com/nnstreamer/api.git to access ml-api-common.h, you need to remove the duplicated ml-api-common.h file in nntrainer.git.  I previously thought that you are using ml-api-common.h of nntrainer.git, not api.git, while you are declaring the dependency w/ pkgconfig.

**Comment by kparichay** [comment]
> > > You may need to update `meson_options.txt` so that its default behavior becomes compatible with non Tizen machines using api/capi/include/platform/ml-api-common.h.
> > 
> > 
> > I am not sure I understand your concern correctly.
> > For non-tizen machines (like debian), the dependency defaults to `capi-ml-common` which is the pkgconfig name for ml-api-common package. Can you please elaborate your concern?
> 
> If you are using capi-ml-common.pc of ml-api-common-dev debian package of github.com/nnstreamer/api.git to access ml-api-common.h, you need to remove the duplicated ml-api-common.h file in nntrainer.git. I previously thought that you are using ml-api-common.h of nntrainer.git, not api.git, while you are declaring the dependency w/ pkgconfig.
> 
> Duplicated header files are dangerous especially if both files do NOT incur build errors.

My bad, I missed it. Updated the PR to remove the duplicated ml-api-common.h in nntrainer.git.
Thanks for the explanation.

**Review by myungjoo — APPROVED** [review]
LGTM

**Comment by kparichay** [comment]
Android build needs to be updated to provide for missing ml-api-common.h

File: jni/Android.mk [hunk 1]
```diff
@@ -82,9 +86,23 @@ LOCAL_SRC_FILES := $(OPENBLAS_ROOT)/lib/libopenblas.a
...
+
 include $(PREBUILT_STATIC_LIBRARY)
 endif #ENABLE_BLAS
 
+## prepare ml common api if nothing present
+ifndef ML_API_COMMON_ROOT
+ifneq ($(MAKECMDGOALS),clean)
+
+ML_API_COMMON_ROOT := $(NDK_INCLUDES_OUT)/ml_api_common
+$(info $(shell ($(NNTRAINER_JNI_ROOT)/prepare_ml-api-common.sh $(ML_API_COMMON_ROOT))))
```
Comment (zhoonit) [inline jni/Android.mk:99]: Did pkgconfig strategy not work?

Comment (kparichay) [reply jni/Android.mk:99]: There is no way to enforce the package to be available when building for jni.

Comment (zhoonit) [reply jni/Android.mk:99]: True that

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**